OpenTrons/opentrons-api
# feat(robot-server): fetch all CSV files used with a protocol recently

Closes AUTH-428

# Overview

Adds a new endpoint `GET /protocols/{protocolId}/dataFiles` which provides a list of all data files used as CSV parameters for the provided protocol. It determines the files used based on the CSV files used and saved in previous analyses. 

# Test Plan

- Upload some data files and use them as csv params for a protocol with CSV params. Use different files in different analysis.
- Check that the new endpoint returns the correct list of data files used

# Review requests

Usual code sanity check

# Risk assessment

Low. New endpoint and isolated feature.
